### ISSUE TITLE
Possible bug: Duplicate "From" fields returning incorrect occurance

### ISSUE THREAD

**Issue by jvendryes** [zbateson/MailMimeParser#74]
Hi,

I came across a scenario where I had multiple `From:` fields. On [3.6 Field Definitions](https://tools.ietf.org/html/rfc5322#section-3.6) you'll see that it specifies a _min_ and _max_ of 1 `From:` field per message.

The parser returned the last `From:` field it found. [RFC 2822 3.6.8](https://tools.ietf.org/html/rfc2822#section-3.6.8) and [RFC 5322 3.6.8](https://tools.ietf.org/html/rfc5322#section-3.6.8]) state:

```
Fields may appear in messages that are otherwise unspecified in this
   document.  They MUST conform to the syntax of an optional-field.
   This is a field name, made up of the printable US-ASCII characters
   except SP and colon, followed by a colon, followed by any text that
   conforms to the unstructured syntax.

   The field names of any optional field MUST NOT be identical to any
   field name specified elsewhere in this document.

...

For the purposes of this specification, any optional field is
   uninterpreted.

```

So in this case, the first occurrence of this field would be the returned value as it's considered an [3.6.2 Originator Field(s)](https://tools.ietf.org/html/rfc5322#section-3.6.2) and the subsequent duplicate fields would be ignored.

Thanks!

**Comment by zbateson** [comment]
Lovely find, thanks @jvendryes

**Comment by zbateson** [comment]
Fixed in 1.0-beta and 0.4.  Will release a new 0.4.11 for it in a few days if nothing comes up :)
